ocean_lib/ocean/ocean_assets.py

```diff
@@ -320,46 +370,9 @@ def create(
             logger.error(msg)
             raise ValueError(msg)
 
-        # Process the DDO
-        asset_dict = asset.as_dictionary()
-        ddo_string = json.dumps(asset_dict, separators=(",", ":"))
-        ddo_bytes = ddo_string.encode("utf-8")
-        ddo_hash = create_checksum(ddo_string)
-
-        # Plain asset
-        if not encrypt_flag and not compress_flag:
-            flags = bytes([0])
-            document = ddo_bytes
-
-        # Only compression, not encrypted
-        elif compress_flag and not encrypt_flag:
-            flags = bytes([1])
-            # Compress DDO
-            document = lzma.compress(ddo_bytes)
-
-        # Only encryption, not compressed
-        elif encrypt_flag and not compress_flag:
-            flags = bytes([2])
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=ddo_string,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
-            )
-            document = encrypt_response.text
-
-        # Encrypted & compressed
-        else:
-            flags = bytes([3])
-            # Compress DDO
-            compressed_document = lzma.compress(ddo_bytes)
-
-            # Encrypt DDO
-            encrypt_response = DataServiceProvider.encrypt(
-                objects_to_encrypt=compressed_document,
-                encrypt_endpoint=f"{provider_uri}/api/services/encrypt",
-            )
-
-            document = encrypt_response.text
+        [document, flags, ddo_hash] = self.process_ddo(
```

Comment:
you don't need the brackets, Python supports multiple returns and will treat them as a tuple by default